Riot/Managers/Room/ChangeMembershipState.swift

```diff
@@ -0,0 +1,56 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// `ChangeMembershipState` represents room membership change states
```

Comment:
Is it really change states?
If yes, all states must show a transition that is happening. All states named `xxxing` are legit. But I cannot see why we have final states in this enum like `joined` or `left`. They do not express a transition.

I am trying to understand the word `change` in `ChangeMembershipState`. Should not be it called jst `RoomMembershipState` ?